### ISSUE TITLE
Zoom in/out of Canvas in editor

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#4022]
## Task Description

Allowing users to zoom in and out on the canvas area was previously mentioned when discussing accessibility requirements for the editor.

## Design
[Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2241%3A2)

## QA

1. Validate that without new experiment enabled, the canvas works as before with one exception: Lasso selection is off by a few pixels. All of this should be working: Dragging in elements, moving and resizing elements, dropping in drop target frames, dragging out of canvas to remove, text edit (also partially outside the canvas), media edit (also partially outside the canvas), canvas resize on browser resize, and more.

2. Validate that with the next experiment enabled, the canvas has a new fancier zoom functionality. This still has missing features to be done in upcoming PR's, but it generally works.

**Comment by swissspidy** [comment]
@dvoytenko Is zooming on the canvas something you've explored originally when working on the canvas implementation? It was mentioned that it might be quite a huge undertaking, so I am curious to hear your thoughts on it.

**Comment by dvoytenko** [comment]
Here's roughly what I discovered:

1. We are in a good shape with the scaling math. It's tested regularly since we already have 3 breakpoints.
1. We should still do zooming with predefined breakpoints. This minimizes rounding errors. A good rule of thumb to look for dimensions that divide in `12`.
1. The selection (including lasso) logic would need to be redone in major ways. However, it should be only a moderate effort. The easy way to look for related problems is to search for `getBoundingClientRect()` in code and retest/correct.
1. The Moveable library has limited scrolling support. Only dragging is supported. Resizing is not supported. E.g. you grab an element and drag to the edge of scrollable area and it scrolls the area as you drag. But grabbing a corner of an element and dragging it to resize past the edge of scrollable area doesn't work. This will affect both: dag/resize/rotate, as well as image/video editor. We'd need to compensate for this.

**Comment by bmattb** [comment]
@o-fernandez following up on the status of the product brief & UX here as this is a significant item that would need to be picked up for sprint 42 starting next Monday for inclusion in v1.3

**Comment by o-fernandez** [comment]
@samitron7 mentioned that she had this spec'd out already in designs from the past, we just need to link to them. @samitron7 can you link to those?

**Comment by samitron7** [comment]
@o-fernandez @barklund  Attached the design in the description as well: [Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2241%3A2).

Zoom design are pretty standardized across most tools so we'll follow similar patterns here. There is a flow for sliders that I have omitted here. We have a visual ticket for sliders and we'll use the same component pattern there.

**Comment by bmattb** [comment]
After discussion with @barklund earlier today, we were advised that this functionality will not be ready in time for the v1.3 release. This was flagged to @o-fernandez and subsequently agreed that this would be deferred to v1.4.0.

**Comment by barklund** [comment]
This is not ready for QA, I just want to get it to a staging environment.

**Comment by swissspidy** [comment]
@barklund You don't need to move it on the board for that 🤔 Even draft PRs are already available on the stories-qa site.

**Comment by barklund** [comment]
> @barklund You don't need to move it on the board for that 🤔 Even draft PRs are already available on the stories-qa site.

My bad, I thought it only worked for QA-ready PR's. Thanks for helping out!

**Comment by csossi** [comment]
@barklund 

1. Verified in QA
2. Only verified here that the the zoom options appear and they indeed zoom to the level selected. Will test all other functionality as existing known tickets/bugs are addressed (i.e. editing while zoomed, etc.)
